frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.jsx

```diff
@@ -199,7 +196,23 @@ function PreviousValueComparison({
           width: getChangeWidth(width),
         })
       : display.percentChange;
-  const separator = <Separator> • </Separator>;
+
+  const separatorColor = isNightMode
```

Comment:
Isn't the separator color also determined by whether it's under `.variation-container-tooltip` (`inTooltip` you mentioned below)

https://github.com/metabase/metabase/blob/252c9423356cfb24315eb5c31a0e97da9a060787/frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.styled.tsx#L64-L66

Shouldn't the condition be like `inTooltip || isNightMode` like line 275?

Reply:
@WiNloSt From my observation, `PreviousValueComparison` is never inside the Tooltip component unlike the other cases where I use `inTooltip || isNightMode`. It appears to be unused code that might've been mistakenly copied over, but I could be wrong.

```
    <Tooltip
      isEnabled={fullDetailDisplay !== fittedDetailDisplay}
      placement="bottom"
      tooltip={
        <Flex align="center">
          <VariationPercent iconSize={TOOLTIP_ICON_SIZE} inTooltip>
            {display.percentChange}
          </VariationPercent>
          <VariationDetails inTooltip>{fullDetailDisplay}</VariationDetails>
        </Flex>
      }
    >
```

Reply:
The `separator` is rendered by `VariationDetails` (line 302) which is rendered under the tooltip like in your code snippet above.

https://github.com/metabase/metabase/blob/c681914357b325f169652c2500b287f5aa7fa690/frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.jsx#L290-L306

Maybe it's also worth to turn `separator` into a separate component that you could pass `inTooltip` flag into just to make the API consistent.

Reply:
@WiNloSt good catch! not sure why I didn't spotted that before 🙏🏻

Reply:
committed the fix [here](https://github.com/metabase/metabase/pull/42347/commits/3d1f6687ef27a71249030aa598eeae3d96971205),